### PR TITLE
Remove docker-app package manager

### PR DESCRIPTION
**Pull request by doanac**
The recent support of pluggable package managers has allowed us to
move the development of the docker-app package manager into our
own source tree.

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
There's a CI failure in aklite testing. Lets hold up on this until the aklite removal is done, and then I'll rebase this change.

File: src/libaktualizr/primary/sotauptaneclient.cc [hunk 1]
```diff
@@ -546,8 +546,7 @@ void SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
       // TODO(OTA-4939): Unify this with the check in
       // PackageManagerFake::fetchTarget() and make it more generic.
       if (primary_ecu_serial == ecu_serial) {
-        if (!target.IsOstree() &&
-            (config.pacman.type == PACKAGE_MANAGER_OSTREE || config.pacman.type == PACKAGE_MANAGER_OSTREEDOCKERAPP)) {
+        if (!target.IsOstree() && config.pacman.type == PACKAGE_MANAGER_OSTREE) {
```
Comment (patrickvacek) [inline src/libaktualizr/primary/sotauptaneclient.cc:549]: How do you deal with this in aktualizr-lite? Obviously we have the TODO/ticket to improve this, but we also have some (internal/private) code that has to work around this type of issue.

**Review by patrickvacek — COMMENTED** [review]
> There's a CI failure in aklite testing. Lets hold up on this until the aklite removal is done, and then I'll rebase this change.

Yep, makes sense.

Again, I think it's cool that we can do this and I'm glad it works for you. As suggested in https://github.com/advancedtelematic/aktualizr/pull/1763#pullrequestreview-496169625, let's think about if there are any missing tests around the API. Also, we should update the changelog.

Also, I still see this on your branch:
```
$ git grep -i dockerapp src/
src/libaktualizr/primary/sotauptaneclient.h:  FRIEND_TEST(DockerAppManager, DockerAppStandalone);
src/libaktualizr/primary/sotauptaneclient.h:  FRIEND_TEST(DockerAppManager, DockerAppBundles);
```

File: src/libaktualizr/primary/sotauptaneclient.cc (hunk 1 above)
Comment (doanac) [reply src/libaktualizr/primary/sotauptaneclient.cc:549]: We don't go through that code path in our code base.

**Comment by patrickvacek** [comment]
@doanac Can you rebase when you get a chance? This looks fine but CI isn't happy as discussed.

**Comment by mike-sul** [comment]
@doanac I am closing this PR in favor of this one https://github.com/advancedtelematic/aktualizr/pull/1775

**Pull request closed without merging**